amun/dockerfile.py

```diff
@@ -73,10 +76,42 @@ def _write_file_string(content: str, path: str) -> str:
     # TODO: handle it in nice way so we can see it nicely in OpenShift's configuration
     content = content.replace('"', '\\"').replace('\n', '\\n\\\n')
     path = path.replace('"', '\"')
-    return f'RUN echo -e "{content}" >"{path}"\n'
+    return f'RUN echo -e "{content}" > "{path}"\n'
+
+
+def _format_dockerfile(dockerfile: str) -> str:
+    """Format and prepare the Dockerfile for a Workflow injection."""
+    def _pipeline(steps: list):
+        return lambda obj: reduce(lambda f, g: g(f), steps, obj)
+
+    def _escape_double_quotes(stream: str) -> str:
+        return re.sub(r'([^\\])"', r'\1\\"', stream)
+
+    def _escape_nested_single_quotes(stream: str) -> str:
+        return re.sub(r"\\'", r"\\\\'", stream)
+
+    def _escape_nested_double_quotes(stream: str) -> str:
+        return re.sub(r'\\"', r'\\\\"', stream)
+
+    def _unescape_single_quotes(stream: str) -> str:
+        return re.sub(r"\\\\'", r"'", stream)
+
+    def _strip(stream: str) -> str:
+        return stream.strip("'")
 
+    pipe = _pipeline([
+        repr,
+        _escape_nested_single_quotes,
+        _escape_nested_double_quotes,
+        _escape_double_quotes,
+        _unescape_single_quotes,
+        _strip,
+    ])
 
-def create_dockerfile(specification: dict) -> tuple:
+    return pipe(dockerfile)
```

Comment:
Any reason why obfuscate code this way? Commands are executed sequentially, IMHO no need to create a "pipeline" for that - comments before each action can be sufficient.

Also note the performance impact of using `re` for a simple string replacement.

Reply:
No reason, other than a constant tendency to overcomplicate things unnecessarily. Thanks for the pointers :)